metadatastore/test/test_mds.py

```diff
@@ -1,11 +1,401 @@
+        A sanitized mongoengine document
+    """
+    with pytest.raises(KeyError):
+        document['id']
+
+
+def setup_syn(mds, custom=None):
+    if custom is None:
+        custom = {}
+    data_keys = {k: {'source': k,
+                     'dtype': 'number',
+                     'shape': None} for k in 'ABCEDEFGHIJKL'
+                 }
+    scan_id = 1
+
+    # Create a BeginRunEvent that serves as entry point for a run
+    rs = mds.insert_run_start(scan_id=scan_id, beamline_id='testing',
+                              time=ttime.time(),
+                              uid=str(uuid.uuid4()),
+                              **custom)
+
+    # Create an EventDescriptor that indicates the data
+    # keys and serves as header for set of Event(s)
+    e_desc = mds.insert_descriptor(data_keys=data_keys,
+                                   time=ttime.time(),
+                                   run_start=rs, uid=str(uuid.uuid4()))
+    return rs, e_desc, data_keys
+
+
+def syn_data(data_keys, count):
+
+    all_data = deque()
+    for seq_num in range(count):
+        data = {k: float(seq_num) for k in data_keys}
+        timestamps = {k: ttime.time() for k in data_keys}
+
+        _time = ttime.time()
+        uid = str(uuid.uuid4())
+        all_data.append({'data': data, 'timestamps': timestamps,
+                         'seq_num': seq_num, 'time': _time,
+                         'uid': uid})
+    return all_data
+
+
+@pytest.mark.parametrize('ver', (0, 1))
+def test_pickle(ver):
     md = MDS(config={'host': 'portland'}, version=1)
     md2 = pickle.loads(pickle.dumps(md))
 
     assert md.version == md2.version
     assert md.config == md2.config
+
+
+def test_event_descriptor_insertion(mds_all):
+    mds = mds_all
+    # format some data keys for insertion
+    data_keys = {'some_value': {'source': 'PV:pv1',
+                                'shape': [1, 2],
+                                'dtype': 'array'},
+                 'some_other_val': {'source': 'PV:pv2',
+                                    'shape': [],
+                                    'dtype': 'number'},
+                 'data_key3': {'source': 'PV:pv1',
+                               'shape': [],
+                               'dtype': 'number',
+                               'external': 'FS:foobar'}}
+
+    document_insertion_time = ttime.time()
+    run_start_uid = mds.insert_run_start(scan_id=3022013,
+                                         beamline_id='testbed',
+                                         owner='tester',
+                                         group='awesome-devs',
+                                         project='Nikea',
+                                         time=document_insertion_time,
+                                         uid=str(uuid.uuid4()))
+    time = ttime.time()
+    # test insert
+    ev_desc_uid = mds.insert_descriptor(run_start_uid, data_keys, time,
+                                        str(uuid.uuid4()))
```

Comment:
Every document had a time. The times on the start, stop, and descriptor documents aren't necessarily scientifically important or precise -- they just tell us wen the document was inserted. The timestamps field in an event document (and, to some extent, the time field in an event document) are more scientifically meaningful.